Documentation/btrfs-subvolume.rst

```diff
@@ -263,6 +263,9 @@ snapshot [-r] [-i <qgroupid>] <source> <dest>|[<dest>/]<name>
 
         -r
                 Make the new snapshot read only.
+        -R
+                Recursively snapshot subvolumes beneath the source. This option cannot be
```

Comment:
This sounds like limitation but we can't do better in user space. This is possible in kernel but is quite complicated so we must exclude -r and -R for now. As a suggestion to documenation, the subvolume can be mad read-only afterwards by changing the property to `ro`.

And as this is using libbtrfsutil the same limitations apply (and need to be documented): it's not atomic and depends on the capabilities to enumerate the subvolumes. Similar to what the recursive deletion has, we might need a special section in documentation for that so we can refer to it  instead of repeating the whole text.

Reply:
Thanks, I've added a bit to the man page.

I might also look into fixing the kernel so that this can be done properly.

Reply:
This still does not mention the limitations, nor referes to a section where it would be explained.